int-tests/server_controller.py

```diff
@@ -1,21 +1,49 @@
 import subprocess
-import os.path
 import sys
 import functools
+import os.path as path
+from threading import Thread
+from queue import Queue, Empty
 
-_module_dir = os.path.abspath(os.path.join(os.path.dirname(__file__)))
-_root_dir = os.path.abspath(os.path.join(_module_dir, '..'))
+module_dir = path.abspath(path.join(path.dirname(__file__)))
+_root_dir = path.abspath(path.join(module_dir, '..'))
 
 
+class StdOutReader:
+  def __init__(self, stream, verbose=False):
```

Comment:
we needed a nonblocking reader to know when the server is ready to accept connections